SU2_CFD/src/output_structure.cpp

```diff
@@ -18989,3 +19002,2268 @@ void COutput::SpecialOutput_AnalyzeSurface(CSolver *solver, CGeometry *geometry,
+
+      iVar = 0;
+
+      /*--- Get the conservative variables for this particular DOF. ---*/
+
+      const su2double *U = solDOFs + j*nVar_First;
+
+      /*--- Load the coordinate values of the solution DOFs. ---*/
+
+      const su2double *coor = volElem[l].coorSolDOFs.data() + j*nDim;
+      for(unsigned short k=0; k<nDim; ++k) {
+        Local_Data[jPoint][iVar] = coor[k];
+        iVar++;
+      }
+
+      /* Load the conservative variables. */
+
+      for(jVar=0; jVar < nVar_First; ++jVar) {
+        Local_Data[jPoint][iVar] = U[jVar];
+        iVar++;
+      }
+
+      /*--- Prepare the primitive states. ---*/
+
+      const su2double DensityInv = 1.0/U[0];
+      su2double vel[3], Velocity2 = 0.0;
+      for(iDim=0; iDim<nDim; ++iDim) {
+        vel[iDim] = U[iDim+1]*DensityInv;
+        Velocity2 += vel[iDim]*vel[iDim];
+      }
+      su2double StaticEnergy = U[nDim+1]*DensityInv - 0.5*Velocity2;
+      DGFluidModel->SetTDState_rhoe(U[0], StaticEnergy);
+
+      /*--- Load data for the pressure, temperature, Cp, and Mach variables. ---*/
+
+      Local_Data[jPoint][iVar] = DGFluidModel->GetPressure(); iVar++;
+      Local_Data[jPoint][iVar] = DGFluidModel->GetTemperature(); iVar++;
+      Local_Data[jPoint][iVar] = DGFluidModel->GetCp(); iVar++;
+      Local_Data[jPoint][iVar] = sqrt(Velocity2)/DGFluidModel->GetSoundSpeed(); iVar++;
+
+      /*--- New variables can be loaded to the Local_Data structure here,
+       assuming they were registered above correctly. ---*/
+
+      if (Kind_Solver == FEM_NAVIER_STOKES){
+        Local_Data[jPoint][iVar] = DGFluidModel->GetLaminarViscosity(); iVar++;
+      }
+      if ((Kind_Solver == FEM_LES) && (config->GetKind_SGS_Model() != IMPLICIT_LES)){
+        // todo: Export Eddy instead of Laminar viscosity
+        Local_Data[jPoint][iVar] = DGFluidModel->GetLaminarViscosity(); iVar++;
+      }
+
+      /*--- Increment the point counter. ---*/
+
+      jPoint++;
+    }
+  }
+
+}
+
+void COutput::PrepareOffsets(CConfig *config, CGeometry *geometry) {
+
+  unsigned long iPoint;
+
+  /*--- Bool to distinguish between the FVM and FEM solvers. ---*/
+
+  unsigned short KindSolver = config->GetKind_Solver();
+  bool fem_solver = ((KindSolver == FEM_EULER) ||
+                     (KindSolver == FEM_NAVIER_STOKES) ||
+                     (KindSolver == FEM_RANS) ||
+                     (KindSolver == FEM_LES));
+
+  /*--- Reset point sorting counters ---*/
+
+  nGlobalPoint_Sort = 0;
+  nLocalPoint_Sort  = 0;
+
+  /*--- Prepare the offsets for the FV solver ---*/
+
+  if (!fem_solver) {
+
+    /*--- Search all send/recv boundaries on this partition for any periodic
+     nodes that were part of the original domain. We want to recover these
+     for visualization purposes. ---*/
+
+    unsigned long iVertex;
+    bool isPeriodic;
+
+    Local_Halo_Sort = new int[geometry->GetnPoint()];
+    for (iPoint = 0; iPoint < geometry->GetnPoint(); iPoint
```

Comment:
We can come back to this very soon, so that we can sort the conn and write ParaView binary in parallel during runtime (and Tecplot binary soon too, hopefully).

Reply:
@economon, do you want to handle this in the current PR or can we do this after the merge? I am afraid I am not too familiar with that part of the code.

Reply:
After the merge - I can take care of this once everything is integrated in develop.